Sources/JsonModel/ResultData/BranchNodeResult.swift

```diff
@@ -135,8 +135,9 @@ public struct PathMarker: Hashable, Codable {
 
 /// Abstract implementation to allow extending an assessment result while retaining the serializable type.
 open class AbstractBranchNodeResultObject : AbstractResultObject {
-    private enum CodingKeys : String, OrderedEnumCodingKey {
+    private enum CodingKeys : String, OrderedEnumCodingKey, OpenOrderedCodingKey {
         case stepHistory, asyncResults, path
+        var relativeIndex: Int { 2 }
```

Comment:
Curious what relativeIndex is used for, and why it's 2 here but 1 in multiple other classes...?

Reply:
relativeIndex is used for open classes where the keys are defined in subclasses. This says that the sort order ranking of this *set* of keys ["stepHistory", "asyncResults", "path"] should be sorted at position 2 which allows other keys to be in position 0 or 1 and therefore ordered before these ones. 

In this example, the keys that are ordered first are the ones in `AbstractResultObject` where `relativeIndex = 0`, next are the keys in `AbstractAssessmentResultObject` where `relativeIndex=1` and finally these keys go at the bottom. It's a super-hacky work-around for not being able to encode keys in the order that I *want* to encode them to make them more readable.